### PR TITLE
Roberto/fix

### PR DESCRIPTION
**Pull request by cipollone**
## Proposed changes

More a discussion about my changes. Let's say it's a draft.
So, the first issue is about pylogics dependency being installed in dev0 instead of alpha0 both from poetry on this repository and on repos that depend from logaut. My adding with --allow-prereleases and installing pylogics from github, it picks the correct version. Let me know if this sounds ok, or I'll keep this as a local workaround.

The second issue is the regex matching with lydia backend. I have installed lydia from latest docker image. The output seems to produce newlines as "\r\n" and the regex fails because it checks for "\n". Also $ fails, because it would be equivalent to "\n" on unix I suppose. Let me know also about that.

### CODE REVIEW

File: logaut/backends/lydia/_lydia_utils.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def postprocess_lydia_output(output: str) -> str:
     :return: the output associated to the DFA.
     """
     regex = re.compile(
-        r"(?<=Computed automaton:\n).*(?=\n\[2)", flags=re.MULTILINE | re.DOTALL
+        r"(?<=Computed automaton:\r\n).*(?=^\[2)", flags=re.MULTILINE | re.DOTALL
```
Comment (marcofavorito) [inline logaut/backends/lydia/_lydia_utils.py:49]: doesn't this break the one without docker? Maybe add `?` after `\r`
```suggestion
        r"(?<=Computed automaton:\r?\n).*(?=^\[2)", flags=re.MULTILINE | re.DOTALL
```

File: pyproject.toml [hunk 2]
```diff
@@ -33,8 +33,8 @@ include = []
 
 [tool.poetry.dependencies]
 python = "^3.7"
-pylogics = "^0.1.0a0"
 pythomata = "^0.3.2"
+pylogics = {git = "https://github.com/whitemech/pylogics", rev = "main", allow-prereleases = true}
```
Comment (marcofavorito) [inline pyproject.toml:37]: Thanks, although it won't be forever

File: logaut/backends/lydia/_lydia_utils.py (hunk 1 above)
Comment (marcofavorito) [reply logaut/backends/lydia/_lydia_utils.py:49]: another solution to the problem is to understand why `\r\n` on Docker; it shouldn't be the case

Comment (cipollone) [reply logaut/backends/lydia/_lydia_utils.py:49]: You're right. Maybe this would be a better idea. I'm happy to see that also logaut has tests.

Comment (cipollone) [reply logaut/backends/lydia/_lydia_utils.py:49]: I'll double check with the docker image

**Comment by cipollone** [comment]
Outdated

**Pull request closed without merging**